src/mysql_vm_helpers.py

```diff
@@ -163,6 +188,31 @@ def wait_until_mysql_connection(self) -> None:
         if not os.path.exists(MYSQLD_SOCK_FILE):
             raise MySQLServiceNotRunningError()
 
+    def connect_mysql_exporter(self) -> None:
+        """Set up mysqld-exporter config options.
+
+        Raises
+            snap.SnapError: if an issue occurs during config setting or restart
+        """
+        cache = snap.SnapCache()
+        mysqld_exporter = cache[MYSQL_EXPORTER_SNAP_NAME]
+
+        try:
+            mysqld_exporter.set(
+                {
+                    "mysql.host": self.instance_address,
```

Comment:
The exporter should work via socket, not the external IP.
Reason - Exporters fetch huge amounts of data, which creates a parasitic monitoring load.
to reduce the effect - avoid transferring via a network.
```suggestion
                    "mysql.host": "unix(/var/run/mysqld/mysqld.sock)",
```